### PR TITLE
artifacts-common Revert Nuget Service Connection Overwrite

### PR DESCRIPTION
**Pull request by embetten**
Ensure nuget service connections allow to overwrite when an endpoint has been previously set.

### CODE REVIEW

**Review by AlexVTor — APPROVED** [review]
LGTM

**Comment by embetten** [comment]
> LGTM

This is not quite right - brb

**Review by embetten — COMMENTED** [review]
Currently appending to array, does not properly replace element

File: common-npm-packages/artifacts-common/credentialProviderUtils.ts [hunk 1]
```diff
@@ -161,48 +161,34 @@ export async function configureCredProviderForSameOrganizationFeeds(protocol: Pr
...
+    var newConnections = buildExternalFeedEndpoints(serviceConnections);
+    var map = new Map(newConnections.map(item => [item.endpoint, item]));
+    const configuredEndpoints = tl.getVariable(CRED_PROVIDER_EXTERNAL_ENDPOINTS_ENVVAR);
+    
+    const existingCredentialsArray = configuredEndpoints ? JSON.parse(configuredEndpoints)['endpointCredentials'] as EndpointCredentials[] : [];
+    existingCredentialsArray.forEach(cred => {
+        if(!map.has(cred.endpoint)){
+            map.set(cred.endpoint, cred);
         }
+    });
```
Comment (AlexVTor) [inline common-npm-packages/artifacts-common/credentialProviderUtils.ts:185]: I notice we aren't warning anymore. Do we want to warn or inform the user that we are not setting creds for an endpoint? If we decide we want to, this looks like a good spot for it

Comment (embetten) [reply common-npm-packages/artifacts-common/credentialProviderUtils.ts]: added it back

**Review by AlexVTor — APPROVED** [review]
Looks good! I like the new use of the mapping and cleaning up the junk around the endpoint container.

**Pull request merged**